lms/djangoapps/teams/static/teams/js/views/edit_team.js

```diff
@@ -0,0 +1,174 @@
+;(function (define) {
+'use strict';
+
+define(['backbone',
+        'underscore',
+        'gettext',
+        'js/views/fields',
+        'teams/js/models/team',
+        'text!teams/templates/edit-team.underscore'],
+       function (Backbone, _, gettext, FieldViews, TeamModel, edit_team_template) {
+           return Backbone.View.extend({
+
+               maxTeamNameLength: 255,
+               maxTeamDescriptionLength: 300,
+
+               events: {
+                   "click .action-primary": "createTeam",
+                   "click .action-cancel": "cancelTeam"
+               },
+
+               initialize: function(options) {
+                   this.courseId = options.teamParams.courseId;
+                   this.teamsUrl = options.teamParams.teamsUrl;
+                   this.topicId = options.teamParams.topicId;
+                   this.languages = options.teamParams.languages;
+                   this.countries = options.teamParams.countries;
+
+                   _.bindAll(this, "cancelTeam", "createTeam");
+
+                   this.teamNameField = new FieldViews.TextFieldView({
+                       model: new TeamModel(),
+                       title: gettext("Team Name (Required) *"),
+                       valueAttribute: 'name',
+                       bindEvents: false,
+                       helpMessage: gettext("A name that identifies your team (maximum 255 characters).")
```

Comment:
@frrances, yes "cannot change after team has been created" applies to all fields including the optional ones. It would be great to have a space where we could have text that applies to all fields.